packages/gestalt/src/SheetMobile/PrimaryAction.js

```diff
@@ -0,0 +1,50 @@
+// @flow strict
+import { type Node, type ElementConfig } from 'react';
+import Button from '../Button.js';
+import Link from '../Link.js';
+
+type Props = {|
+  accessibilityLabel: string,
+  href?: string,
+  label: string,
+  onClick?: $ElementType<ElementConfig<typeof Button>, 'onClick'>,
+  rel?: $ElementType<ElementConfig<typeof Link>, 'rel'>,
+  size?: $ElementType<ElementConfig<typeof Button>, 'size'>,
+  target?: $ElementType<ElementConfig<typeof Link>, 'target'>,
+|};
+
+export default function ToastPrimaryAction({
+  accessibilityLabel,
+  href,
+  label,
+  onClick,
+  rel,
+  size = 'lg',
+  target,
+}: Props): Node {
+  if (href && label)
+    return (
+      <Button
+        accessibilityLabel={accessibilityLabel}
+        href={href}
+        rel={rel}
+        target={target}
```

Comment:
seems like the only changes here are href, ref, target and role

Purely stylistic, but maybe, you can just return one button , and pass the rest as properties. I think smth like this might be allowed

```
const linkProperties = {}
if(href){
linkProperties = {role:"link", href, rel, target}
}

<Button accessbilityLabel={blah} ....{linkProperties}>
```

Reply:
I don't like spreading props as we wont be able to use codemods if needed. It's more like an internal rule, no spreading.

Reply:
For Flow usually better using conditional . Also we'll be breaking Button into Button / ButtonLink, sthis will be easier to migrate

Reply:
Cool, I didn't know. Just thought spreading was concise, but also am used to typescript flows soooo, sounds good!